File: 118-pascals-triangle/pascals-triangle.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>();
            row.add(1); // First element is always 1

            // Fill in the middle elements
            for (int j = 1; j < i; j++) {
                int prevRowVal = triangle.get(i - 1).get(j - 1) + triangle.get(i - 1).get(j);
                row.add(prevRowVal);
            }

            if (i > 0) {
                row.add(1); // Last element is always 1 (except for first row)
            }

            triangle.add(row);
        }

        return triangle;
    }
}

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows):
        triangle = [[1]]  # Start with the first row

        for i in range(1, numRows):
            row = [1]
            prev_row = triangle[-1]
            for j in range(1, len(prev_row)):
                row.append(prev_row[j - 1] + prev_row[j])
            row.append(1)
            triangle.append(row)
        
        return triangle
